config/config.go

```diff
@@ -310,6 +319,15 @@ func GetBool(key string, defaultValue bool) (value bool) {
 	return viper.GetBool(key)
 }
 
+func GetFirstInt(defaultValue int, keys ...string) (value int) {
```

Comment:
Do we really need these functions?

Can't we keep it simple like this? https://github.com/rudderlabs/rudder-server/blob/581aef4bc02c06d99a7f1f637d150391bb294988/router/config.go#L7

Reply:
Simple, it depends one your perspective. Repeating this piece of code everywhere required won't keep it simple or safe. IMO it is simpler to provide functions that encapsulate such common logic, tested once, used everywhere.

Reply:
I do find the introduction of `GetFirst...` methods a bit too much, and it could make readability and understanding harder. Especially in emergencies when you try to understand which configuration is applied. Having many ways to define a configuration can confuse.

After thinking about passing, `HttpClient.timeout` is not helpful. Every client has its own needs and needs to be configured separately. It also makes sense to have a default value hardcoded as `30sec` for most, but changing the underlying base value for all HTTP clients has no use case (other than keeping backwards compatibility).

So, my suggestion is to drop the `GetFirst...` and rename `HttpClient.timeout` to `HttpClient.<client-name>.timeout`. A good idea to do it now, before introducing v1.0.0, as this is going to break existing behaviour for configurations.

Reply:
e.g. isn't this simpler?

```
func getRouterConfigBool(key, destType string, defaultValue bool) bool {
	return config.GetFirstBool(defaultValue, "Router."+destType+"."+key, "Router."+key)
}

func getRouterConfigInt(key, destType string, defaultValue int) int {
	return config.GetFirstInt(defaultValue, "Router."+destType+"."+key, "Router."+key)
}

```

Reply:
@chandumlg are you ok with what @lvrach is proposing?

Reply:
yeah sounds good.